pkg/daemon/cleanup/cleanup.go

```diff
@@ -143,7 +161,8 @@ func mergeServices(s1, s2 []servicescm.Service) []servicescm.Service {
 }
 
 // removeServices uses the given manager to remove all the given Windows services from this instance.
-func removeServices(svcMgr manager.Manager, services []servicescm.Service) error {
+func removeServices(latestCMData, versionCMData *servicescm.Data, svcMgr manager.Manager) error {
+	services := getServicesToRemove(latestCMData, versionCMData)
```

Comment:
This is overloading the responsibilities of this function

Reply:
It was in deconfigure where it was only used to get the services to be removed. I could move it back but I think it is better suited where it is used instead of passing it in.